library/components/TDropdown.vue

```diff
@@ -52,6 +56,12 @@
         },
 		mounted() {
 			onClickOutside(this.$refs.dropdownFilter, (event) => {
+				// Temporary solution
+                // If a component is appending it's inner dropdown directly to body.
+                for (let cl of this.outsideElementClasses) {
+                    // If an outside element is still open, keep the dropdown from closing.
+                    if (document.querySelector(`.${cl}`)) return;
+                } 
```

Comment:
Did the calendar stay open, when the dropdown closed?

Reply:
When you click outside, it will first close calendar, then another click will close dropdown.

Reply:
That's with this change, but what about earlier? Why did you need this? Was it because the dropdown closed and the calendar stayed open?

Reply:
Yes, the reason behind this is that (as I understand), this date picker is appended directly to body when it is opened, instead of being a child of dropdown, it is a direct second child of `<body />`, so with this fix we make sure that whatever classes we specify dropdown to watch before closing are checked before it is collapsed.

Reply:
Because of this, @click.stop also does not work here.

Reply:
😬 ah, ok. I was just curious. I can't think of a better solution right now, but I want to understand fully.